### DISCUSSION TITLE
Problem with running lammps with frozen model

### DISCUSSION THREAD

**Discussion by 1119609754** [deepmodeling/deepmd-kit#2537] [Q&A] [answered]
When I submit the command to run the task, it stops quickly, but no error message is given and the log file is empty.
I don't know if it's the in.lammps or the model?My model, which was derived with ovito, has 1.5 million atoms.

the error:
root@bohrium-15775-1018178:/data/lammps# lmp -i in.lammps
Warning:
This LAMMPS executable is in a conda environment, but the environment has
not been activated. Libraries may fail to load. To activate this environment
please see https://conda.io/activation.
LAMMPS (23 Jun 2022 - Update 1)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
Loaded 1 plugins from /opt/deepmd-kit-2.2.1/lib/deepmd_lmp
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
Loaded 1 plugins from /opt/deepmd-kit-2.2.1/lib/deepmd_lmp
Reading data file ...
  orthogonal box = (-104.5248 -60.3468 -147.8256) to (113.2352 65.3757 160.1444)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  1500000 atoms
  read_data CPU = 2.479 seconds
DeePMD-kit WARNING: Environmental variable TF_INTRA_OP_PARALLELISM_THREADS is not set. Tune TF_INTRA_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable TF_INTER_OP_PARALLELISM_THREADS is not set. Tune TF_INTER_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable OMP_NUM_THREADS is not set. Tune OMP_NUM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
Summary of lammps deepmd module ...
  >>> Info of deepmd-kit:
  installed to:       /opt/deepmd-kit-2.2.1
  source:             v2.2.1
  source branch:       HEAD
  source commit:      3ac8c4c7
  source commit at:   2023-03-16 12:33:24 +0800
  surpport model ver.:1.1 
  build variant:      cuda
  build with tf inc:  /opt/deepmd-kit-2.2.1/include;/opt/deepmd-kit-2.2.1/include
  build with tf lib:  /opt/deepmd-kit-2.2.1/lib/libtensorflow_cc.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
  >>> Info of lammps module:
  use deepmd-kit at:  /opt/deepmd-kit-2.2.1DeePMD-kit WARNING: Environmental variable TF_INTRA_OP_PARALLELISM_THREADS is not set. Tune TF_INTRA_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable TF_INTER_OP_PARALLELISM_THREADS is not set. Tune TF_INTER_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable OMP_NUM_THREADS is not set. Tune OMP_NUM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit: Successfully load libcudart.so
2023-05-18 14:08:00.335628: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2 AVX AVX2 AVX512F FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-05-18 14:08:00.345750: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/cuda/lib64:
2023-05-18 14:08:00.345777: W tensorflow/stream_executor/cuda/cuda_driver.cc:269] failed call to cuInit: UNKNOWN ERROR (303)
2023-05-18 14:08:00.345851: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (bohrium-15775-1018178): /proc/driver/nvidia/version does not exist
2023-05-18 14:08:00.347285: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.
2023-05-18 14:08:00.386011: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:354] MLIR V1 optimization pass is not enabled
  >>> Info of model(s):
  using   1 model(s): graph.pb 
  rcut in model:      4.2
  ntypes in model:    1
  using compute id:        
System init for delete_atoms ...
Generated 0 of 0 mixed pair_coeff terms from geometric mixing rule
Neighbor list info ...
  update every 1 steps, delay 10 steps, check yes
  max neighbors/atom: 2000, page size: 100000
  master list distance cutoff = 6.2
  ghost atom cutoff = 6.2
  binsize = 3.1, bins = 71 41 100
  2 neighbor lists, perpetual/occasional/extra = 1 1 0
  (1) command delete_atoms, occasional, copy from (2)
      attributes: full, newton on
      pair build: copy
      stencil: none
      bin: none
  (2) pair deepmd, perpetual
      attributes: full, newton on
      pair build: full/bin/atomonly
      stencil: full/bin/3d
      bin: standard
Deleted 0 atoms, new total = 1500000
1500000 atoms in group newton

CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE

Your simulation uses code contributions which should be cited:
- USER-DEEPMD package:
The log file lists these citations in BibTeX format.

CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE

WARNING: Using 'neigh_modify every 1 delay 0 check yes' setting during minimization (src/min.cpp:187)
Generated 0 of 0 mixed pair_coeff terms from geometric mixing rule
Neighbor list info ...
  update every 1 steps, delay 0 steps, check yes
  max neighbors/atom: 2000, page size: 100000
  master list distance cutoff = 7.2
  ghost atom cutoff = 7.2
  binsize = 3.6, bins = 61 35 86
  1 neighbor lists, perpetual/occasional/extra = 1 0 0
  (1) pair deepmd, perpetual
      attributes: full, newton on
      pair build: full/bin/atomonly
      stencil: full/bin/3d
      bin: standard
Setting up cg style minimization ...
  Unit style    : metal
  Current step  : 0
/opt/deepmd-kit-2.2.1/bin/lmp: line 11:    93 Killed                  /opt/deepmd-kit-2.2.1/bin/_lmp "$@"

my in.lammps:
clear 
units             metal 
dimension         3 
boundary          p p p
atom_style        atomic
timestep        0.001
read_data  work.lmp 
mass     1  12.01 

pair_style    deepmd graph.pb
pair_coeff    * *
delete_atoms overlap 0.825 all all
neighbor 3.0 bin
neigh_modify delay 10 check yes  

region        newton  block  INF INF INF INF INF INF units box
group         newton   region newton

compute eng all pe/atom
compute 3 all ke/atom 
compute eatoms all reduce sum c_eng

reset_timestep 0
timestep  0.001

thermo     10
thermo_style   custom step pe lx ly lz press pxx pyy pzz
thermo_modify  lost ignore flush yes
dump  1  all cfg 1000 dump.diamond-minimization_*.cfg mass type xs ys zs c_eng
dump_modify   1  element C

min_style cg
minimize   1e-15 1e-15 50000 50000

undump 1

reset_timestep	0
timestep 0.001
fix		 1 all  npt temp 250 250 0.1 aniso 1 1 1 

velocity 		 all create 300 12345 mom yes rot no
 

thermo 1000
thermo_style  custom step lx ly lz press pxx pyy pzz pe temp
thermo_modify lost ignore flush yes

dump 		1 all cfg 1000 dump.relax_*.cfg mass type xs ys zs fx fy fz
dump_modify 1 element C

run 3000
unfix  1

undump 1

fix 1 newton npt temp 250 300 0.1 aniso 1 1 1       
      
       
dump                1 all cfg 10000 dump.loading+x_*.cfg mass type xs ys zs vx vy vz c_eng c_3 

thermo   100
thermo_style    custom step temp vol press pxx pyy pzz c_eatoms pe ke etotal lx ly lz
thermo_modify  lost ignore flush yes

run          100000
unfix        1
undump   1

**Comment by njzjz** [comment, marked as answer]
The message said that the process was killed by the operating system. Usually it means the memory is exhausted. See https://stackoverflow.com/a/726762

**Reply by 1119609754** [reply]
Thank you for your answer to solve the problem
But I had another problem:

WARNING: Using 'neigh_modify every 1 delay 0 check yes' setting during minimization (src/min.cpp:187)
Generated 0 of 0 mixed pair_coeff terms from geometric mixing rule
Neighbor list info ...
  update every 1 steps, delay 0 steps, check yes
  max neighbors/atom: 2000, page size: 100000
  master list distance cutoff = 7.2
  ghost atom cutoff = 7.2
  binsize = 3.6, bins = 61 35 86
  1 neighbor lists, perpetual/occasional/extra = 1 0 0
  (1) pair deepmd, perpetual
      attributes: full, newton on
      pair build: full/bin/atomonly
      stencil: full/bin/3d
      bin: standard
ERROR: DeePMD-kit Error: TensorFlow Error: RESOURCE_EXHAUSTED: OOM when allocating tensor with shape[1,3240000000] and type double on /job:localhost/replica:0/task:0/device:CPU:0 by allocator mklcpu
	 [[{node ProdEnvMatA}]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info. This isn't available when running in Eager mode.
 (/home/conda/feedstock_root/build_artifacts/libdeepmd_1663923207577/work/source/lmp/pair_deepmd.cpp:418)
Last command: minimize   1e-15 1e-15 50000 50000

I found the cause of the problem: too many atoms,My model has 1.5 million atoms,So many atoms are used to eliminate boundary effects,How many atoms do I need to reduce appropriately for deepmd parallel lammps simulation?

**Reply by njzjz** [reply]
It should depend on how much memory there is.
